token/cli/src/main.rs

```diff
@@ -964,80 +917,54 @@ async fn command_transfer(
             } else {
                 true
             }
-        } else {
+        }
+        // otherwise trust the cli flag
+        else {
             fund_recipient
         };
 
-        if needs_funding {
+        // and now we determine if we will actually fund it, based on its need and our willingness
+        let fundable_owner = if needs_funding {
             if fund_recipient {
-                if !config.sign_only {
-                    minimum_balance_for_rent_exemption += config
-                        .program_client
-                        .get_minimum_balance_for_rent_exemption(Account::LEN)
-                        .await?;
-                    println_display(
-                        config,
-                        format!(
-                            "  Funding recipient: {} ({} SOL)",
-                            recipient_token_account,
-                            lamports_to_sol(minimum_balance_for_rent_exemption)
-                        ),
-                    );
-                }
-                instructions.push(create_associated_token_account(
-                    &config.fee_payer.pubkey(),
-                    &recipient,
-                    &mint_info.address,
-                    &mint_info.program_id,
-                ));
+                println_display(
+                    config,
+                    format!("  Funding recipient: {}", recipient_token_account,),
+                );
+
+                Some(recipient)
             } else {
                 return Err(
                     "Error: Recipient's associated token account does not exist. \
                                     Add `--fund-recipient` to fund their account"
                         .into(),
                 );
             }
-        }
+        } else {
+            None
+        };
+
+        (recipient_token_account, fundable_owner)
+    };
+
+    // set up memo if provided...
+    if let Some(text) = memo {
+        token.with_memo(text, vec![config.default_signer.pubkey()]);
     }
 
-    if use_unchecked_instruction {
-        #[allow(deprecated)]
-        instructions.push(transfer(
-            &mint_info.program_id,
-            &sender,
-            &recipient_token_account,
-            &sender_owner,
-            &config.multisigner_pubkeys,
-            transfer_balance,
-        )?);
-    } else {
-        instructions.push(transfer_checked(
-            &mint_info.program_id,
+    // ...and, finally, the transfer
+    let res = token
+        .transfer(
             &sender,
-            &mint_info.address,
             &recipient_token_account,
             &sender_owner,
-            &config.multisigner_pubkeys,
             transfer_balance,
-            mint_info.decimals,
-        )?);
-    }
-    if let Some(text) = memo {
-        instructions.push(spl_memo::build_memo(
-            text.as_bytes(),
-            &[&config.fee_payer.pubkey()],
-        ));
-    }
```

Comment:
the old version pushed the memo _after_ the transfer instruction, but the client (and, consequently, my version) pushes it _before_. based on this in the token22 processor:
```
        if memo_required(&destination_account) {
            check_previous_sibling_instruction_is_memo()?;
        }
```

the previous implementation was in error